src/librustc/hir/lowering.rs

```diff
@@ -71,13 +71,13 @@ pub struct LoweringContext<'a> {
 
 pub trait Resolver {
     // Resolve a global hir path generated by the lowerer when expanding `for`, `if let`, etc.
-    fn resolve_generated_global_path(&mut self, path: &hir::Path, is_value: bool) -> Def;
+    fn resolve_hir_path(&mut self, path: &hir::Path, is_value: bool) -> PathResolution;
```

Comment:
FWIW, changes in this module conflict with #37676, so you can either revert them to merge earlier, or wait for that PR (which would be merged already if it weren't for windows).
Also keep in mind that `PathResolution` doesn't really matter after HIR lowering in #37676, but YMMV.